src/features/share/asImage/ShareAsImagePreferences.ts

```diff
@@ -0,0 +1,100 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { db } from "../../../services/db";
+import { AppDispatch, useAppDispatch, useAppSelector } from "../../../store";
+import { useCallback, useEffect } from "react";
+import { DeepPartial } from "../../../helpers/deepPartial";
+
+export interface ShareAsImagePreferences {
+  post: {
+    hideCommunity: boolean;
+    hideUsername: boolean;
+    watermark: boolean;
+  };
+  comment: {
+    includePostDetails: boolean;
+    includePostText: boolean;
+    hideUsernames: boolean;
+    watermark: boolean;
+    allParentComments: boolean;
+  };
+}
+
+const initialState: ShareAsImagePreferences = {
+  post: {
+    hideCommunity: false,
+    hideUsername: false,
+    watermark: false,
+  },
+  comment: {
+    includePostText: false,
+    includePostDetails: false,
+    hideUsernames: false,
+    watermark: false,
+    allParentComments: false,
+  },
+};
+
+const { reducer, actions } = createSlice({
+  name: "shareAsImagePreferences",
+  initialState,
+  reducers: {
+    setShareAsImagePreferences: (
+      state,
+      action: PayloadAction<DeepPartial<ShareAsImagePreferences>>,
+    ) => {
+      const { post, comment } = action.payload;
+      state.post.hideCommunity =
+        post?.hideCommunity ?? state.post.hideCommunity;
+      state.post.hideUsername = post?.hideUsername ?? state.post.hideUsername;
+      state.post.watermark = post?.watermark ?? state.post.watermark;
+
+      state.comment.includePostText =
+        comment?.includePostText ?? state.comment.includePostText;
+      state.comment.includePostDetails =
+        comment?.includePostDetails ?? state.comment.includePostDetails;
+      state.comment.hideUsernames =
+        comment?.hideUsernames ?? state.comment.hideUsernames;
+      state.comment.watermark = comment?.watermark ?? state.comment.watermark;
+      state.comment.allParentComments =
+        comment?.allParentComments ?? state.comment.allParentComments;
+
+      db.setSetting(
+        "share_as_image_preferences",
+        JSON.parse(JSON.stringify(state)),
```

Comment:
Alternative to this was 
```suggestion
        {
        post: { ...state.post },
        comment: { ...state.comment },
      }
```